fuzzers/eclipser/fuzzer.py

```diff
@@ -11,88 +11,181 @@
-        'foo',  # Specifies how command line argument is passed, just a file.
-        '-f',
+        eclipser_out,
+        '--arg',  # Specifies the command-line of the program.
+        'foo',
+        '-f',  # Specifies the path of file input to fuzz.
         'foo',
-        '--maxfilelen',
-        # Default is too low (8 bytes), match experiment config at:
-        # https://github.com/SoftSec-KAIST/Eclipser-Artifact/blob/6aadf02eeadb0416bd4c5edeafc8627bc24ebc82/docker-scripts/experiment-scripts/package-exp/run_eclipser.sh#L25
-        '1048576',
-        # Default is low (0.5 sec), recommended to use higher:
-        # https://github.com/google/fuzzbench/issues/70#issuecomment-596060572
+        '-v',  # Controls the verbosity.
+        '2',
         '--exectimeout',
         '5000',
     ]
-    if os.listdir(input_corpus):  # Important, otherwise Eclipser crashes.
+    if os.listdir(input_corpus):  # Specify inputs only if any seed exists.
         command += ['-i', input_corpus]
-    print('[fuzz] Running command: ' + ' '.join(command))
+    print('[eclipser] Run Eclipser with command: ' + ' '.join(command))
     subprocess.Popen(command)
 
-    process = Process(target=copy_corpus_directory,
-                      args=(
-                          encoded_temp_corpus,
-                          output_corpus,
-                      ))
-    process.start()
-
-
-def copy_corpus_directory(encoded_temp_corpus, output_corpus):
-    """Copies corpus periodically from encoded corpus directory into output
-  directory."""
-    while True:
-        # Wait for initial fuzzer initialization, and after every copy.
-        time.sleep(120)
-
-        subprocess.check_call([
-            'dotnet',
-            '/Eclipser/build/Eclipser.dll',
-            'decode',
-            '-i',
-            os.path.join(encoded_temp_corpus, 'testcase'),
-            '-o',
-            output_corpus,
-        ])
+
+def afl_master(input_corpus, output_corpus, target_binary):
+    """Run AFL master instance."""
+    command = [
+        './afl-fuzz',
+        '-i',
+        input_corpus,
+        '-o',
+        output_corpus,
+        # Use no memory limit as ASAN doesn't play nicely with one.
+        '-m',
+        'none',
+        # Use same default 1 sec timeout, but add '+' to skip hangs.
+        '-t',
+        '1000+',
+        '-M',
+        'afl-master',
+    ]
+    dictionary_path = utils.get_dictionary_path(target_binary)
+    if dictionary_path:
+        command.extend(['-x', dictionary_path])
+    command += [
+        '--',
+        target_binary,
+        # Pass INT_MAX to afl the maximize the number of persistent loops it
+        # performs.
+        '2147483647'
+    ]
+    print('[afl_master] Run AFL master with command: ' + ' '.join(command))
+    output_stream = subprocess.DEVNULL
+    subprocess.check_call(command, stdout=output_stream, stderr=output_stream)
+
+
+def afl_worker(input_corpus, output_corpus, target_binary):
```

Comment:
Can you please use afl_fuzzer.run_afl_fuzz to avoid code duplication with additional_flags option (that is also important as we sometimes update the helper for corner cases).

Reply:
(Same as above)

Reply:
See comment above, run_afl_fuzz can be fixed.